sympy/geometry/parabola.py

```diff
@@ -420,3 +422,61 @@ def vertex(self):
         else:
             vertex = self.axis_of_symmetry.intersection(self)[0]
         return vertex
+
+    #functin to check if point is on parabola
+    def contain(self, p):
+        """Check if the point `p` is on the parabola."""
+        if not isinstance(p, Point): #check if p is a Point instance
+            p = Point(p)
+        x, y = symbols('x y')
```

Comment:
You should use Dummy symbols so as to avoid using symbols that may already be in use in the Parabola. Isit possible that simplification will not reduce the expression to 0? Perhaps `return eq.subs({...}).equals(0)` should be used.